ansible_base/lib/abstract_models/common.py

```diff
@@ -123,6 +125,29 @@ def get_summary_fields(self):
                     response[field.name] = getattr(self, field.name).summary_fields()
         return response
 
+    def _get_reverse_view(self, field):
+        # A field can specify a related_view to use for the reverse lookup
+        # This looks like:
+        #
+        # foo = models.ManyToManyField(...)
+        # foo.related_view = 'foo-list'
+        #
+        # It can also specify related_view = None to opt out of showing up here.
+        #
+        # The default is to try to guess the view name based on the model and field name.
+        # But this won't always work in every case, so this gives models a way to specify.
+        reverse_view = getattr(field, 'related_view', False)
+
+        if reverse_view is None or reverse_view:
+            # Give fields a chance to opt out of showing up here by forcing related_view to None
+            return reverse_view
+
+        # otherwise reverse_view is False, meaning we should try to guess the view name
+        if isinstance(field, models.ManyToManyField):
+            return f"{underscore(self.__class__.__name__)}-{field.name}-list"
+
+        return f"{underscore(field.name)}-detail"
```

Comment:
This line eats at me. Previously with `reverse_view = f"{underscore(field.related_model.__name__)}-detail"` in place of this, we could have relied on the guess instead of adding customization.

You've made really good progress in DAB standardizing models so that we don't need to write extra view and serializer logic for everything, and I'd like to continue that. I think this goes in the wrong direction.